src/display/annotation_layer.js

```diff
@@ -724,18 +728,34 @@ class PopupElement {
       popup.style.backgroundColor = Util.makeCssRgb(r | 0, g | 0, b | 0);
     }
 
-    let contents = this._formatContents(this.contents);
     let title = document.createElement('h1');
     title.textContent = this.title;
+    popup.appendChild(title);
+
+    // The modification date is shown in the popup instead of the creation
+    // date if it is available and can be parsed correctly, which is
+    // consistent with other viewers such as Adobe Acrobat.
+    try {
+      const dateObject = PDFDateString.toDateObject(this.modificationDate);
```

Comment:
Please refer to the comment in the `toDateObject` method.

Reply:
Fixed in the new commit.